### ISSUE TITLE
SystemBroadcastReceiver throws SecurityException on Android 14

### ISSUE THREAD

**Issue by mikejeep** [bugsnag/bugsnag-android#1870]
### Describe the bug
In a project with the target set to API 34 (Android 14) on an Android 14 device, a SecurityException is thrown because RECEIVER_EXPORTED or RECEIVER_NOT_EXPORTED is not specified when receiver is being registered in BugSnag code that appears to be in `SystemBroadcastReceiver.kt`.

This issue is described in the Android 14 behavior changes [here](https://developer.android.com/about/versions/14/behavior-changes-14):
> "Runtime-registered broadcasts receivers must specify export behavior: Apps and services that target Android 14 and use [context-registered receivers](https://developer.android.com/guide/components/broadcasts#context-registered-receivers) are required to specify a flag to indicate whether or not the receiver should be exported to all other apps on the device: either RECEIVER_EXPORTED or RECEIVER_NOT_EXPORTED, respectively. This requirement helps protect apps from security vulnerabilities by leveraging the [features for these receivers introduced in Android 13](https://developer.android.com/about/versions/13/features#runtime-receivers)."

### Steps to reproduce
1. Add BugSnag to Android project
~~2. Enable StrictMode with `penaltyLog()`~~ (Discovered that StrictMode is not required to reproduce issue)
3. Launch app
4. Check Logcat output

### Environment
* Android version: 14 Beta 4
* Bugsnag version: 5.30.0
* Emulator or physical device: Google Pixel 7

### Logcat error message
```java
Failed to register receiver
java.lang.SecurityException: com.me.myapp: One of RECEIVER_EXPORTED or RECEIVER_NOT_EXPORTED should be specified when a receiver isn't being registered exclusively for system broadcasts
at android.os.Parcel.createExceptionOrNull(Parcel.java:3057)
at android.os.Parcel.createException(Parcel.java:3041)
at android.os.Parcel.readException(Parcel.java:3024)
at android.os.Parcel.readException(Parcel.java:2966)
at android.app.IActivityManager$Stub$Proxy.registerReceiverWithFeature(IActivityManager.java:5684)
at android.app.ContextImpl.registerReceiverInternal(ContextImpl.java:1852)
at android.app.ContextImpl.registerReceiver(ContextImpl.java:1792)
at android.app.ContextImpl.registerReceiver(ContextImpl.java:1780)
at android.content.ContextWrapper.registerReceiver(ContextWrapper.java:755)
at com.bugsnag.android.ContextExtensionsKt.registerReceiverSafe(ContextExtensions.kt:24)
at com.bugsnag.android.SystemBroadcastReceiver$Companion.register(SystemBroadcastReceiver.kt:28)
at com.bugsnag.android.SystemBroadcastReceiver.register(Unknown Source:2)
at com.bugsnag.android.Client$3.run(Client.java:348)
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:487)
at java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
at java.lang.Thread.run(Thread.java:1012)
```
</details>

**Comment by johnkiely1** [comment]
Hi @mikejeep, thanks for raising. we going to take a look at addressing this.

**Comment by clr182** [comment]
Hi @mikejeep yes this has been released in [v5.31.0](https://github.com/bugsnag/bugsnag-android/releases/tag/v5.31.0). Thanks again for raising this issue.
